lib/slots.ts

```diff
@@ -1,3 +1,6 @@
+import { Temporal } from '.';
```

Comment:
I suspect that consistently only importing types (not implementations) works better. Not 100% sure, but I think I've seen cases where mixing the two causes weird loops in type resolution that the TS compiler complained about. 

```suggestion
import type { Temporal } from '..';
```

Reply:
I've been running into this issue all the time lately, where vscode suggests one or the other import and the two aren't assignable to each other. It would be great to try and 'exclude' temporal.ts (the actual module file) from being an import suggestion so that only the typings would show up instead.

Reply:
Yeah, if you find out how to do that let me know! Very annoying.